lib/resque/worker.rb

```diff
@@ -861,7 +862,11 @@ def perform_with_fork(job, &block)
 
       begin
         @child = fork do
-          unregister_signal_handlers if term_child
+          if term_child
+            unregister_signal_handlers
+          else
+            @signal_handler.reopen
```

Comment:
It seems difficult to create tests for handling signals, but let me try it later.